src/tv2_afvd_showstyle/migrations/util.ts

```diff
@@ -105,3 +107,102 @@ export function getOutputLayerDefaultsMigrationSteps(versionStr: string): Migrat
+		version,
+		canBeRunAutomatically: true,
+		validate: (context: MigrationContextShowStyle) => {
+			const oldConfigTable = (context.getBaseConfig(oldTableId) as unknown) as TableConfigItemValue
+			if (!oldConfigTable || oldConfigTable.length === 0) {
+				return false
+			}
+
+			const newConfigTable = (context.getBaseConfig(newTableId) as unknown) as TableConfigItemValue
+			return !newConfigTable
+		},
+		migrate: (context: MigrationContextShowStyle) => {
+			const oldConfigTable = (context.getBaseConfig(oldTableId) as unknown) as TableConfigItemValue
+			const newConfigTable = ((context.getBaseConfig(newTableId) as unknown) as TableConfigItemValue) ?? []
+			oldConfigTable.map(value => newConfigTable.push(value))
+
+			context.setBaseConfig(newTableId, newConfigTable)
+			context.removeBaseConfig(oldTableId)
+		}
+	}
+}
+
+/**
+ * "Renames" the name of a Blueprint configuration by overriding the value of the old configuration onto the new configuration - then removes the old configuration
+ */
+export function renameBlueprintConfiguration(
+	version: string,
+	oldConfigurationName: string,
+	newConfigurationName: string
+): MigrationStepShowStyle {
+	return {
+		id: `${version}.rename.blueprint.configuration.${oldConfigurationName}.to.${newConfigurationName}`,
+		version,
+		canBeRunAutomatically: true,
+		validate: (context: MigrationContextShowStyle) => {
+			const oldConfig = (context.getBaseConfig(oldConfigurationName) as unknown) as BasicConfigItemValue
+			if (!oldConfig || Object.keys(oldConfig).length === 0) {
+				return false
+			}
+
+			const newConfig = (context.getBaseConfig(newConfigurationName) as unknown) as BasicConfigItemValue
+			return !newConfig
+		},
+		migrate: (context: MigrationContextShowStyle) => {
+			const oldConfig = (context.getBaseConfig(oldConfigurationName) as unknown) as BasicConfigItemValue
+
+			context.setBaseConfig(newConfigurationName, oldConfig)
+			context.removeBaseConfig(oldConfigurationName)
+		}
+	}
+}
+
+/**
+ *  For all variants: "Renames" the blueprint configuration by overriding the value of the old configuration onto the new configuration - then removes the old configuration
+ */
+export function renameBlueprintsConfigurationForAllVariants(
+	version: string,
+	oldConfigName: string,
+	newConfigName: string
+): MigrationStepShowStyle {
+	return {
+		id: `${version}.rename.blueprint.configuration.${oldConfigName}.to.${newConfigName}.for.all.variants`,
+		version,
+		canBeRunAutomatically: true,
+		validate: (context: MigrationContextShowStyle) => {
+			const allVariants: IBlueprintShowStyleVariant[] = context.getAllVariants()
+
+			if (allVariants.length === 0) {
+				return false
+			}
+
+			const noVariantsHaveOldConfig: boolean = allVariants.every((variant: IBlueprintShowStyleVariant) => {
+				const oldConfig = context.getVariantConfig(variant._id, oldConfigName)
+				return !oldConfig || Object.keys(oldConfig).length === 0
+			})
+
+			return !noVariantsHaveOldConfig
```

Comment:
You check if noVariantsHaveOldConfig, and then you negate it, which in result logically turns into to _some variants have_..., so wouldn't this be sufficient?
```suggestion
			const someVariantsHaveOldConfig: boolean = allVariants.some((variant: IBlueprintShowStyleVariant) => {
				const oldConfig = context.getVariantConfig(variant._id, oldConfigName)
				return oldConfig !== undefined
			})

			return someVariantsHaveOldConfig
```

Reply:
I think it was because of the `Object.keys(oldConfig).length === 0` which is no longer necessary.

Reply:
Updated